openedx-unsupported/devstack
# build: Make a test.in file and upgrade dependencies (foundation for tests)

This allows us to keep the tests in our usual language rather than adding a new one. (Thanks Jeremy for finding this library!)

I've also moved the CLI tests to earlier in the CI run so that we get faster feedback and don't have to wait for cloning first. Really, they should be their own workflow file, but that's work for another day.

There is also a foundational commit for adding a `test.in` and upgrading deps before adding anything new.

----

I've completed each of the following, or confirmed they do not apply to this PR:

    - Already tested on: Linux
    - Testing instructions:
        - These are run by CI, so you should see if they pass that way. But if you'd like to test that they can fail properly, try changing the spawn command in each test to be `make help` or something else that won't behave as the test expects it to. You should get an EOF or timeout failure.
        - There are preconditions for running one of the tests, but it will tell you what they are. :-)
